prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchemaReader.kt

```diff
@@ -0,0 +1,62 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7.schema
+
+import com.fasterxml.jackson.databind.JsonMappingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.JacksonYAMLParseException
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import org.apache.logging.log4j.kotlin.Logging
+import java.io.File
+import java.io.InputStream
+
+/**
+ * Read schema configuration.
+ */
+object ConfigSchemaReader : Logging {
+    /**
+     * Read a schema [schemaName] from a file given the root [folder].
+     * @return the validated schema
+     * @throws Exception if the schema is invalid
+     */
+    fun fromFile(schemaName: String, folder: String? = null): ConfigSchema {
+        val rawSchema = readSchemaTreeFromFile(schemaName, folder)
+
+        if (!rawSchema.isValid()) {
+            throw Exception("Invalid schema $schemaName: \n${rawSchema.errors.joinToString("\n")}")
+        }
+        return rawSchema
+    }
+
+    /**
+     * Read a complete schema tree from a file for [schemaName] in [folder].
+     * @return the validated schema
+     */
+    internal fun readSchemaTreeFromFile(schemaName: String, folder: String? = null): ConfigSchema {
```

Comment:
Please update comment to include this is recursive, so that anyone not familiar with the functionality has less confusion.

Reply:
Will add comment in the next issue to minimize merge conflicts